services/queue/src/artifacts.js

```diff
@@ -356,7 +356,11 @@ let replyWithArtifact = async function(taskId, runId, name, req, res) {
         skipCache = true;
       }
 
-      if (skipCache && skipCDN) {
+      if (this.signPublicArtifactURLs) {
```

Comment:
I think this might read a bit better if we swapped the preexisting `if`/`else if` since we don't actually use any cloudmirror/`skipCache`/`skipCDN` logic if `signPublicArtifactURLs` is true. I'm imagining as follows (I think this expresses the same logic, feel free to push back if not):

```js
if (this.signPublicArtifactURLs || bucket === this.privateBucket.bucket) {
  url = await this.privateBucket.createSignedGetUrl(/*...*/);
} else if (bucket === this.publicBucket.bucket) {
  // no diff here
}
```

Reply:
I nearly went this way but I figured I'd have to factor out the bucket object as its own variable (anything else seemed too messy). That would end up having much more impact on the code.

I do agree with the readability argument though, I'll have a go.

Reply:
I think I've addressed this now - though I'm not amazingly happy about picking the bucket object. There's technically an edge-case where the specified bucket in the artifact isn't either of the public/private buckets which would be handled differently in this case (but I don't think that's possible to get into without deployment config changes).